sympy/solvers/ode.py

```diff
@@ -1218,8 +1218,7 @@ class in it.  Note that a hint may do this anyway if
                 matching_hints[s + "_Integral"] = r
                 if undetcoeff['test']:
                     r['trialset'] = undetcoeff['trialset']
-                    matching_hints["nth_linear_constant_coeff_undetermined_"
-                        "coefficients"] = r
+                    matching_hints["nth_linear_constant_coeff_undetermined_coefficients"] = r
```

Comment:
It's important for people who use the horizontal space for windows next to each other, so in general it's one of the more important guidelines. (Full disclosure: I'm one of those.)
That said, sometimes I do break it, too. In this case, I'd put the string on a line of its own to make it stick out less to the right, but I'd still keep it in a single string even if it sticks out to the right, because in this case, searchability indeed trumps horizontal space economics.

One could consider putting the string as a global variable. That would keep the uses shorter, and it would eliminate possible typos. It would also create a place where the meaning of each hint could be documented.
One could rightfully argue that this kind of refactoring belongs into a separate PR though.